src/PantheonTerminalWindow.vala

```diff
@@ -992,6 +990,8 @@ namespace PantheonTerminal {
             }
 
             name_check_timeout_id = Timeout.add (50, () => {
+                save_opened_terminals ();
```

Comment:
I am not sure this is necessary here.  It is already being called after adding, closing and restoring a tab.  Only need to save if tab labels are changed (see below) due to disambiguation. Maybe at line 1033?

Reply:
It will not in all cases update the saved tabs on a name change with that set up unfortunately.